vault/audited_headers.go

```diff
@@ -183,7 +183,13 @@ func (a *AuditedHeadersConfig) invalidate(ctx context.Context) error {
 }
 
 // ApplyConfig returns a map of approved headers and their values, either hmac'ed or plaintext.
+// If the supplied headers are empty or nil, the input value is returned as-is.
 func (a *AuditedHeadersConfig) ApplyConfig(ctx context.Context, headers map[string][]string, salter audit.Salter) (result map[string][]string, retErr error) {
+	// Return early if we don't have headers.
+	if headers == nil || len(headers) < 1 {
+		return headers, nil
```

Comment:
This could lead to a panic. Because without this code, it's **result** that is returned and it is never `nil`. Even if the **headers** parameter was `nil`. I would recommend changing this line to `return map[string][]string{}, nil`